evalit/_base.py

```diff
@@ -76,29 +115,68 @@ def __init__(
             config = self.load_yaml(config)
         if not isinstance(config, dict):
             raise TypeError(
-                f"Invalid type for config. Expected Dict[str, str]. Got {type(config)}"
+                (
+                    f"Invalid type for config. Expected Dict[str, str]. "
+                    f"Got {type(config)}"
+                )
             )
         self.config = copy.deepcopy(config)
         if debug:
-            logger.debug(self.__get_redacted_cfg())
+            logger.debug(self._get_redacted_cfg(self.config))
         self._sanity_check_config(self.config)
 
     @abstractmethod
     def run_automation(self, **kwargs) -> Tuple[TransferDTO]:
         """
-        Main method to start transfer
+        Main entrypoint/method to start transfer
+
+        Args:
+            ```kwargs```: ```Any```
+                extra keyword arguments that can be used
+                for the transfer process
+
+        Returns:
+            Tuple of `structures.TransferDTO` objects
         """
         raise NotImplementedError()
 
     @staticmethod
     def load_yaml(config_yaml: TYPE_PATH) -> Dict[str, str]:
         """
         Load config from yaml file!
+
+        Args:
+            ```config_yaml```: ```Union[str, pathlib.Path]```
+                Path to YAML file to be loaded
+
+        Returns:
+            ```Dict[str, str]``` dictionary config
+
+        Example of dictionary structure:
+
+            .. code-block: python
+
+                {
+                    "source_token": <username>,
+                    "source_secret": <password>,
+                    "source_s3_endpoint": "http://127.0.0.1:<PORT_1>",
+                    "source_s3_bucket": "src",
+                    "source_s3_region": "us-east-1",
+                    "dest_token": <username>,
+                    "dest_secret": <password>,
+                    "dest_s3_endpoint": "htt*://127.0.0.1:<PORT_2>",
+                    "dest_s3_bucket": "dest",
+                    "dest_s3_region": "us-east-1",
```

Comment:
would it be better to have `{ 'source': { 'token': ...}, 'dest': { 'token': }}` format?